### DISCUSSION TITLE
signup flow is confusing

### DISCUSSION THREAD

**Discussion by ekeren** [winglang/wing#745] [General]
# Description 

Wing is still not ready for the masses, the product is in a very early stage and we want to be able to work on it with the right people on our side. 

Our approach is that opening this project for all, will just cause noise and will be counter productive. 

# Problem

The current signup flow requires filling in a form for enrolling to alpha and authenticating a GitGub app (See appendix > Current Journeys)

## Proposed solution 

This reduce frustration, we propose to remove the links to docs.winglang.io from the landing page as long as the project is closed. Folks will be able to sign up to get access and then they will be able to access our docs.

## Future

Either remove the sign up well or change sign up to only require a github authentication.

---

# Appendix 

## Current Journeys
Currently people are confused and complaining, rightfully so, about the signup flow. 
There are a couple of optional journeys 

### Journey 1: Starts with request alpha access
1. A visitor user goes into winglang.io 
1 . He clicks “request alpha access” fill in the form
1. He gets added to the queue 
1. (after some time) Gets invited 
1. Goes to github.com/winglang/wing → README → docs.winglang.io 
1. 😠 Requested to approves the github auth app  → approves
1. 🙂 Has access to everything 

### Journey 2: Starts with request alpha access but looks for more before he gets approved
1. An anonymous user goes into winglang.io 
1. He clicks “request alpha access” fill in the form
1. He gets added to the queue 
1. Goes to docs.winglang.io 
1. 😠 Surprisingly Requested to approves the github auth app  → approves
1. 😡 Still no access 
1. (after some time) Gets invited 
1. 🙂 Has access to everything 

## Journey 3: User starts with looking for more info
1. An anonymous user goes into winglang.io 
1. Goes to docs.winglang.io 
1. 😠 Surprisingly requested to approves the github auth app  → approves
1. 😡 Still need to complete the form
1. Added to queue
1. (after some time) Gets invited 
1. 🙂 Has access to everything 

## Proposed Journey

1. Visitor goes into winglang.io 
   1. Visitor don’t see docs.winglang.io links
   1. Visitor don’t see github.com/winglang/wing links
1. Visitor clicks the CTA “request alpha access” 
1. (visitor → user) The visitor authenticate a github application with permission: email, github, name (instead of typeform-1) 
1. The user gets a “thanks for registering to our waiting queue email”
1. The user gets approve
1. The user has access to everything
1. The user still need to log into his github account in order to

**Comment by ShaiBer** [comment]
I'm not so sure about the last part of the description:
> We believe that opening this project for all, will just cause noise and will be counter productive

We currently accept everyone and it seems to be working well for us.
So I think we should consider just opening the repo up and then we can really simplify people's lives (also remove the step in the readme about the PAT)

**Reply by ekeren** [reply]
First, I think this is a valid proposal. 

This is the main downside as I see it: 

Currently there is not a lot to do in the project in terms of retention, meaning that I assume most people will have no reason to keep interacting with the project after the first couple of days engaging with it. This off course will change in the future

Further more, because we can only retain the extreme tech enthusiastic, this project right now will be too un-mature for most people that decided to join the alpha.  

If we open it for all, we will not have a way to update these people on what is new and is happening with the project.

**Reply by ShaiBer** [reply]
Valid concern, but maybe there are other ways we can go about it? we can encourage people to register to receive notification when we enter our beta and they can actually build something with it. Also, if our PR is good enough when we go public they will see it since they are probably in close circles to us if they are here now.
By adding hurdles in people's way to get started, we risk lowering the already small number of tech enthusiasts that we manage to capture in order to optimize our ability to recover a small part of our later target audience that might stop interacting after initial signup

**Reply by Chriscbr** [reply]
@ShaiBer I think in another discussion you made a good point about the value of holding back on some forms of news so that we can still make noise and announcements in the future. I think making the project public would be a useful milestone to keep for a future announcement (many months from now, probably). I also don't think making the repo public today would help us build the Wing ecosystem faster (or better). It is possible to drown a plant by overwatering.

**Reply by ShaiBer** [reply]
> It is possible to drown a plant by overwatering.


Haha @Chriscbr this is very funny, and true.

However, right now I feel that we are drowning in complications that are a result of us being private:
- cost of GitHub seats
- cost of mergify seats
- various bugs and issues that were opened and dealt with to make the authorization mechanism work
- user experience issues with the signup flow
- resentment by some of it's very existence 
- being unable to link into our repo
- And most importantly a 50% drop in the funnel from people who requested access to people who authorized with GitHub, which means that we're losing a lot of people that we worked hard to bring and could help us shape the future of Wing.

Therefore, I think making the repo public actually will help us build the Wing ecosystem fasterz and saving this for a future PR opportunity will not be worth the cost.

But I agree that we shouldn't rush into things, especially since I don't have a solution to how to deal with too much water as you put it, so I'm for implementing @ekeren's solution and revisit the going public decision later. But not much later :)

**Reply by Chriscbr** [reply]
I see - thanks for listing out those pains/complications. Individually they don't look so bad, but they definitely add up...

**Reply by ShaiBer** [reply]
Yes :) and I didn't even list everything: the PAT step that adds a hurdle to the completion of the readme, and probably more

**Comment by skyrpex** [comment]
I like this proposal; anything that simplifies a process is always welcome, IMO.

**Reply by skyrpex** [reply]
Regarding implementation, I'm not sure if Guts (which is in charge of the winglang.io website) will be happy about that.

**Comment by ShaiBer** [comment]
Also discussed with @eladb , let's do @ekeren proposal so we have a consistent user journey, and then we can properly plan how to open up for everyone soon, with the right messaging around it and with verifying that we are really OK with it

**Comment by eladb** [comment]
I've removed all the docs.winglang.io links from winglang.io
